atmo/coordinator/coordinator.go

```diff
@@ -86,49 +86,40 @@ func (c *Coordinator) Start() error {
 		return errors.Wrap(err, "failed to App.Start")
 	}
 
-	// establish connections defined by the app
+	// establish connections defined by the app.
 	c.createConnections()
 
-	caps, err := capabilities.Render(rcap.DefaultCapabilityConfig(), c.App, c.log)
-	if err != nil {
-		return errors.Wrap(err, "failed to renderCapabilities")
-	}
-
-	// we have to wait until here to initialize Reactr
-	// since the appsource needs to be fully initialized
-	if err := c.exec.UseCapabilityConfig(caps); err != nil {
-		return errors.Wrap(err, "failed to UseCapabilityConfig")
-	}
-
 	// do an initial sync of Runnables
-	// from the AppSource into RVG
+	// from the AppSource into RVG.
 	c.SyncAppState()
 
 	return nil
 }
 
-// SetupHandlers configures all of the app's handlers and generates a Vektor Router for the app
+// SetupHandlers configures all of the app's handlers and generates a Vektor Router for the app.
 func (c *Coordinator) SetupHandlers() *vk.Router {
 	router := vk.NewRouter(c.log)
 
-	// set a middleware on the root RouteGroup
+	// set a middleware on the root RouteGroup.
 	router.Before(scopeMiddleware)
 
-	// if in headless mode, enable runnable authentication
+	// if in headless mode, enable runnable authentication.
 	if *c.opts.Headless {
 		router.Before(c.headlessAuthMiddleware())
 	}
 
-	// mount each handler into the VK group
-	for _, h := range c.App.Handlers() {
-		switch h.Input.Type {
-		case directive.InputTypeRequest:
-			router.Handle(h.Input.Method, h.Input.Resource, c.vkHandlerForDirectiveHandler(h))
-		case directive.InputTypeStream:
-			if h.Input.Source == "" || h.Input.Source == directive.InputSourceServer {
-				router.HandleHTTP(http.MethodGet, h.Input.Resource, c.websocketHandlerForDirectiveHandler(h))
-			} else {
-				c.streamConnectionForDirectiveHandler(h)
+	// mount each handler into the VK group.
+	for _, application := range c.App.Applications() {
```

Comment:
I'm a bit torn about this because theoretically different applications could have conflicting routes within them. In Headless mode, that will never happen, but in a normal app, it could.

We have to decide if we care. I'm tempted to not care for now and say "bundle source will never have multiple apps, and everything else is headless anyways", and call it a day, but it's something to think about.

If we choose to leave this as-is, we should file an issue to investigate further; for example we could create a separate VK router for each application, inspect each request as it comes in (perhaps looking at the host header) and then use the appropriate router to handle it. That's super overkill for now, but it could be something to look at later.

Reply:
Hm, would different applications be exposed on ... different domains... yeah, okay... so I think we need routers for each domain.

Reply:
Opened https://github.com/suborbital/atmo/issues/136 to track it